R/utils.R

```diff
@@ -35,8 +35,14 @@ load_csv_files <- function(filenames, optionals, quiet = FALSE, ...) {
       if(!quiet) message("Note: optional input ", f, "not found")
       next
     }
-    suppressMessages(readr::read_csv(fqfn, comment = COMMENT_CHAR, ...)) %>%
-      parse_csv_header(fqfn) %>%
+
+    # Read the file header and extract the column type info from it
+    assert_that(file.exists(fqfn))
+    header <- readLines(fqfn, n = 20)  # only first 20 lines
```

Comment:
20 lines is probably a safe assumption, but if you wanted to make this more robust, you could do something like this:

``` r
# For testing
string <-
"# Description: Something
# Source: something else
# Column types: this
a, b, c, d"
con <- textConnection(string)

# Real code would be:
# con <- file(path, "r")

is_comment <- TRUE

header <- character()

while (is_comment) {
  line <- readLines(con, n = 1)
  is_comment <- grepl("^#", line)
  if (is_comment) {
    header <- c(header, line)
  }
}

header
#> [1] "# Description: Something" "# Source: something else"
#> [3] "# Column types: this"
```

This is the strategy I used to read the YAML header for [`csvy`](https://github.com/leeper/csvy/blob/master/R/get_yaml_header.R) and [`metar`](https://github.com/ashiklom/metar/blob/readr_backend/R/read_yaml_header.R).

Reply:
Yeah, I was being lazy here. Thanks @ashiklom will fix.

Reply:
Only thing I added was `on.exit(close(con))` because otherwise was getting lots of warnings.